lib/chef/knife/ec_restore.rb

```diff
@@ -238,11 +238,17 @@ def upload_org_data(name)
           # pivotal and members of the admins group. Use the same strategy
           # above here.
           #
-          ['admins', 'billing-admins', 'public_key_read_access'].each do |group|
+          groups = ['admins', 'billing-admins']
+          groups.push('public_key_read_access') if server.supports_public_key_read_access?
+
+          groups.each do |group|
             restore_group(chef_fs_config, group, :clients => false)
           end
 
-          ['/acls/groups/billing-admins.json', '/acls/groups/public_key_read_access.json'].each do |acl|
+          acls_groups_paths = ['/acls/groups/billing-admins.json']
+          acls_groups_paths.push('/acls/groups/public_key_read_access.json') if server.supports_public_key_read_access?
```

Comment:
@jeremymv2 

So, some of these issues we can fix and others we can't.  It likely isn't /just/ because of the naming change.

Older versions of Chef Server have no way to reference "global groups".  Because of this, they will never be able to upload a user acl since the user-acl contains global groups. However, depending on what your backup has, this usually isn't a problem since most people don't touch user ACLs and we check if the user-acl needs to be updated before attempting to update it.  Since the user ACL gets created when we create the user, for many backups of older servers, the version on disk matched the newly created user acl.

Unfortunately, the addition of the ::server-admins group and the renaming of the globa_admins groups mean that many more people now have user acls in their backups that don't match the version that the server will create by default.

Here is what I think a full solution would look like:

1) If we are restoring to a version of Chef Server that doesn't support global groups in ACL updates, skip updating any user ACLs that have global groups in them (this is probably ALL user ACLs).

2) If we are restoring to a version of Chef Server that DOES support global groups in ACL updates:

a) Replace instances of ORGNAME_global_admins with ORGNAME_read_access_group before uploading UNLESS ORGNAME_global_admins /also/ exists as a group inside that organizations.

b) Compare the newly created ACL on the server to the version in the backup.  If the user ACL doesn't contain a ::server-admins group but the server supports server-admins, we should probably (perhaps as an option) add this to the ACL.

If after (a) and (b) the ACLs are the same, we can skip the upload entirely.  Some of this would be made easier if the backups included a VERSION file that told us which version of the chef-server a backup came from.

It isn't going to be 100% clean in any direction unfortunately.